src/core/net/ip6.cpp

```diff
@@ -1226,6 +1224,11 @@ Error Ip6::HandleDatagram(Message &aMessage, MessageOrigin aOrigin, const void *
         }
     }
 
+    // never forward reassembled frames as they were already delivered as fragments
+    if (aIsReassembled)
+    {
+        forwardHost = false;
```

Comment:
To understand the issue more, is it possible to have `forwardThread=true` when `aIsReassembled=true` ?

Reply:
I think forwarding to thread should work just fine, the problem is with passing to big frames to host.
For example src/posix/platform/netif.cpp:processReceive has an assert: assert(length <= kMaxIp6Size);